### PR TITLE
Remove duplicate updateBoundParameter calls 

### PR DESCRIPTION
**Pull request by AnHeuermann**
## Related Issues
https://trac.openmodelica.org/OpenModelica/ticket/5775

## Changes
  - Fixes memory leak

## TODO
  - Will probably still lose memory if you use `fmi2Get` during initialization of the FMU.
  - Only reallocate memory if parameters really changed.

### CODE REVIEW

**Comment by AnHeuermann** [comment]
Since we discussed in Slack if it is okay to simply remove the function calls https://github.com/OpenModelica/OpenModelica/pull/6824/files#diff-9635c01d852bad3896d5c39a2e79e9a1b9e358ea733e76d31bc662b8e798dfdfL781

I did a small test:

```modelica
model simple
  parameter String nomTable = "profiltest";
  parameter String nomFichier = "profil.txt";
  Modelica.Blocks.Sources.CombiTimeTable combiTimeTable(fileName = nomFichier, smoothness = Modelica.Blocks.Types.Smoothness.LinearSegments, tableName = nomTable, tableOnFile = true);
  Modelica.Blocks.Interfaces.RealOutput real_output;
  parameter Real a=1;
  parameter Real b=0;
  parameter Real xStart=sin(a*b);
  Real x(fixed=true, start=xStart);
equation
  connect(combiTimeTable.y[1], real_output);
  der(x) = a*x;
end simple;
```

Export as FMU and run with OMSimulator:
```lua
oms_setCommandLineOption("--deleteTempFiles=true")
oms_setTempDirectory("./temp/")
oms_newModel("model")
oms_addSystem("model.root", oms_system_sc)
-- instantiate FMUs
oms_addSubModel("model.root.system1", "simple.fmu")
-- simulation settings
oms_setResultFile("model", "results.mat")
oms_setStopTime("model", 0.1)
oms_setFixedStepSize("model.root", 1e-4)
oms_instantiate("model")
oms_setReal("model.root.system1.b", 10)
print("xStart =",oms_getReal("model.root.system1.xStart"))
oms_initialize("model")
oms_simulate("model")
oms_terminate("model")
oms_delete("model")
````
and got the correct value for `xStart`.

**Comment by AnHeuermann** [comment]
Inside [initialization() ](https://github.com/OpenModelica/OpenModelica/blob/fd748b6da40bbdf3b0ff69e5f47c836ba214e4f2/OMCompiler/SimulationRuntime/c/simulation/solver/initialization/initialization.c#L643) `updateBoundParameters `and `updateBoundVariableAttributes` are still be called. So there is no use in calling these functions right before calling initialization.

```C
if(!(pInitFile && strcmp(pInitFile, ""))) {
    data->callback->updateBoundParameters(data, threadData);
    data->callback->updateBoundVariableAttributes(data, threadData);
  }
```

**Comment by lochel** [comment]
So when do you update bound parameters? In getReal and initialization?

File: OMCompiler/SimulationRuntime/fmi/export/openmodelica/fmu2_model_interface.c.inc [hunk 1]
```diff
@@ -776,10 +774,8 @@ fmi2Status fmi2GetReal(fmi2Component c, const fmi2ValueReference vr[], size_t nv
 
   if (comp->_need_update)
   {
-    if (modelInitializationMode == comp->state)
+    if (comp->state == modelInitializationMode)
```
Comment (lochel) [inline OMCompiler/SimulationRuntime/fmi/export/openmodelica/fmu2_model_interface.c.inc:777]: Why did you move the constant value to the rhs? It was on the lhs on purpose.

Comment (AnHeuermann) [reply OMCompiler/SimulationRuntime/fmi/export/openmodelica/fmu2_model_interface.c.inc:777]: Well the value we are interested in is `comp->state` and that it is in `modelInitializationMode`. So I like to read it first.
Why would you do it the other way around (besides personal preferences)? I can revert it, I don't really care, only while reading the code it was simpler to understand for me.

Comment (lochel) [reply OMCompiler/SimulationRuntime/fmi/export/openmodelica/fmu2_model_interface.c.inc:777]: Constants on the left hand side avoid errors if it unintentionally get assigned instead of compared, i.e. typing = instead of ==. For that reason it is good to just be consistent and put constant on the lhs.

Comment (AnHeuermann) [reply OMCompiler/SimulationRuntime/fmi/export/openmodelica/fmu2_model_interface.c.inc:777]: Okay, I'll revert that change

**Review by lochel — APPROVED** [review]
